### PR TITLE
BGP large community support

### PR DESCRIPTION
**Pull request by czerwonk**
* receiving large communities
* sending large communities
* adding large communities by filter
* filtering for exact large communities

### CODE REVIEW

File: protocols/bgp/packet/path_attributes.go [hunk 1]
```diff
@@ -574,3 +669,16 @@ func isEndOfASSset(asPathPart string) bool {
 func fourBytesToUint32(address [4]byte) uint32 {
 	return uint32(address[0])<<24 + uint32(address[1])<<16 + uint32(address[2])<<8 + uint32(address[3])
 }
+
+func read4BytesAsUin32(buf *bytes.Buffer) (uint32, error) {
```
Comment (hikhvar) [inline protocols/bgp/packet/path_attributes.go]: There is a typo in this function name. It should be read4BytesAsUint32.

To reduce duplicated code we can use this function inside decodeUint32.

Comment (czerwonk) [reply protocols/bgp/packet/path_attributes.go]: Thanks. I will fix that :)

Comment (czerwonk) [reply protocols/bgp/packet/path_attributes.go]: fixed the typo, removed the code duplication and uncluttered the deocdes by passing an string for error string generation to decodeUint32, so we can return instant in decodeXYZ.

Comment (hikhvar) [reply protocols/bgp/packet/path_attributes.go]: Looks nice! :+1:

**Pull request merged**